application/apps/indexer/parsers/src/someip.rs

```diff
@@ -0,0 +1,330 @@
+        Some(service) => {
+            service_name = Some(service.name.clone());
+            match service.get_method(method_id) {
+                Some(method) => {
+                    method_name = Some(method.name.clone());
+                    match message_type {
+                        MessageType::Request
+                        | MessageType::RequestNoReturn
+                        | MessageType::Notification => method.get_request(),
+                        MessageType::Response => method.get_response(),
+                        _ => None,
+                    }
+                }
+                _ => None,
+            }
+        }
+        _ => None,
+    };
+
+    let som_type = match fibex_type {
+        Some(value) => match FibexTypes::build(value) {
+            Ok(result) => Some(result),
+            _ => None,
+        },
+        _ => None,
+    };
+
+    (service_name, method_name, som_type)
+}
+
+fn rpc_payload_string(details: RpcPayloadDetails, payload: &RpcPayload) -> String {
+    let (service_name, method_name, som_type) = details;
+
+    let signature = format!(
+        "{}::{}",
+        match service_name {
+            Some(value) => value,
+            _ => String::from("'Unknown Service'"),
+        },
+        match method_name {
+            Some(value) => value,
+            _ => String::from("'Unknown Method'"),
+        }
+    );
+
+    if payload.is_empty() {
+        return format!("- {}", signature);
+    }
+
+    let content = match som_type {
+        Some(mut value) => {
+            let mut parser = SOMParser::new(payload);
+            match value.parse(&mut parser) {
+                Ok(_) => {
+                    format!("{}", value)
+                }
+                Err(error) => format!("{}", error),
+            }
+        }
+        _ => String::from("'Unknown Type'"),
+    };
+
+    format!("- {} {}", signature, content)
+}
+
+#[derive(Debug, Serialize)]
+pub struct SomeipLogMessage {
+    name: String,
+    description: Option<String>,
+    bytes: Vec<u8>,
+}
+
+impl SomeipLogMessage {
+    pub fn from(name: String, description: Option<String>, bytes: Vec<u8>) -> Self {
+        SomeipLogMessage {
+            name,
+            description,
+            bytes,
+        }
+    }
+}
+
+impl LogMessage for SomeipLogMessage {
+    fn to_writer<W: Write>(&self, writer: &mut W) -> Result<usize, std::io::Error> {
+        let bytes = self.bytes.to_vec();
+        let len = bytes.len();
+        writer.write_all(&bytes)?;
+        Ok(len)
+    }
+}
+
+impl Display for SomeipLogMessage {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            format!(
+                "SOME/IP {}{}",
+                self.name,
+                match &self.description {
+                    Some(description) => format!("\n{}", description),
+                    None => String::from(""),
+                },
+            )
```

Comment:
`write!` already takes a format string, so another `format!` will only create another copy on the heap...needs to be avoided, s.th. like
```
write!("f, "SOME/IP {}{}", self.name, ...)
```

Reply:
done.